datatypes/src/raster/raster_tile.rs

```diff
@@ -11,32 +12,70 @@ use crate::util::Result;
 use num_traits::AsPrimitive;
 use serde::{Deserialize, Serialize};
 
+pub type RasterTile<D, T> = BaseTile<GridOrEmpty<D, T>>;
 pub type RasterTile2D<T> = RasterTile<GridShape2D, T>;
 pub type RasterTile3D<T> = RasterTile<GridShape3D, T>;
 
+pub type MaterializedRasterTile<D, T> = BaseTile<Grid<D, T>>;
+pub type MaterializedRasterTile2D<T> = MaterializedRasterTile<GridShape2D, T>;
+pub type MaterializedRasterTile3D<T> = MaterializedRasterTile<GridShape3D, T>;
+
 /// A `RasterTile2D` is the main type used to iterate over tiles of 2D raster data
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
 #[serde(rename_all = "camelCase")]
-pub struct RasterTile<D, T>
-where
-    D: GridSize + GridSpaceToLinearSpace,
-    T: Pixel,
-{
+pub struct BaseTile<G> {
     /// The `TimeInterval` where this `RasterTile is valid`.
     pub time: TimeInterval,
     /// The tile position is the position of the tile in the gird of tiles with origin at the origin of the global_geo_transform.
     /// This is NOT a pixel position inside the tile.
     pub tile_position: GridIdx2D,
     /// The global geotransform to transform pixels into geographic coordinates
     pub global_geo_transform: GeoTransform,
-    /// The data of the `RasterTile` is stored as `Grid`
-    pub grid_array: Grid<D, T>,
+    /// The data of the `RasterTile` is stored as `Grid` or `NoDataGrid`. `GridOrEmpty` allows a combination of both.
+    pub grid_array: G,
```

Comment:
Why not just use that enum?

Reply:
as explained above, this allows to use the BaseTile for both, the NoData and the Materialized tiles.